src/events/message.js

```diff
@@ -0,0 +1,36 @@
+module.exports = async (client, message) => {
+    const userId = message.author.id
+    
+    // Handle command arguments
+    const args = message.content.slice(prefix.length).trim().split(/ +/g)
+    const cmd = args.shift().toLowerCase()
+    const generalCmds = ['help', 'register', 'total', 'leaderboard', 'ping', 'faq']
+    
+    if (message.author.bot) return
+
+    // Check if the user has an account.
+    const user = await User.findOne({ discordId: userId })
+
+    // Updates user name/discriminator in DB
+    if (user && user.name !== message.author.username) {
+        const updatedName = { name: message.author.username }
+        User.updateOne({ discordId: userId }, updatedName, err => checkErr(err, client, () => {
+            console.log(`Updated username ${user.name} to ${message.author.username}`)
+        }))
+    }
+    
+    // Add user to message reward cooldown
+    if (user && !user.banned && !messageCooldowns.includes(userId)) reward(userId, client)
+    messageCooldowns.push(userId)
```

Comment:
`messageCooldowns` is not defined and is missing a function to clear the array.

Reply:
What do you mean by "_is missing a function to clear the array_"?

Reply:
Originally, there was an interval that would clear the array of user ID's every minute. That way, they are rewarded if their ID is not included in the array.

Reply:
Located [here](https://github.com/TheTypingMatch/le-cash-bot/blob/9d0603c1e56610bc77e5095246ffb31540899892/index.js#L72).

Reply:
I moved that to `ready.js` under Events folder.